### ISSUE TITLE
WebP-lossless decode problem

### ISSUE THREAD

**Issue by tuif** [lovell/sharp#2341]
Are you using the latest version? Is the version currently in use as reported by `npm ls sharp` the same as the latest version as reported by `npm view sharp dist-tags.latest`?
Yes, we installed the latest version.

What are the steps to reproduce?
We have webp-lossless images converted from PNG. We try to read RGB of each pixel.
When we use sharp@0.23.4, the first RGB is reported as 1 213 125.
When we use any other version later than 0.24.0(eg. 0.26.0), the first RGB is reported as 0 212 124.
Since the web-lossless file is from PNG，we know 1 213 125 is the correct result.

What is the expected behaviour?
The latest version could procedure the correct result

Are you able to provide a minimal, standalone code sample, without other dependencies, that demonstrates this problem?
const sharp = require('sharp');
const fs = require('fs');
(async function main() {
  let img = fs.readFileSync(PATH_TO_WEBP);
  let webp = await sharp(img).raw().toBuffer();
  console.log('channels:', webp.length / 256 / 256)
  console.log('first data:', webp[0], webp[1], webp[2]);
})();

Are you able to provide a sample image that helps explain the problem?
Yes, please refer to the attachment. Since github doesnot support webp attachment, we packed it as zip.

More Information about the web file:
1. It's converted from PNG file.
2. We tried different sharp version for converting(0.23.4 vs 0.26.0). They procedure the exact same output.

[terrain.webp.zip](https://github.com/lovell/sharp/files/5139729/terrain.webp.zip)

What is the output of running `npx envinfo --binaries --system`?
  System:
    OS: macOS Mojave 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz
    Memory: 122.27 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.16.0 - /usr/local/bin/node
    npm: 6.14.8 - /usr/local/bin/npm

**Comment by lovell** [comment]
Are you using the prebuilt binaries? If so, the version of libwebp will differ, e.g. sharp v0.23.4 provided libwebp v1.0.3 whereas the latest sharp v0.26.0 provides libweb v1.1.0.

An example of a recent bug fix in libwebp relating to lossless is https://github.com/webmproject/libwebp/commit/0e48d889eb90363c6ebf8ea276bae4d892f9d3c0

You can see which version is in use by inspecting the value of `sharp.versions.webp` - https://sharp.pixelplumbing.com/api-utility#versions

**Comment by lovell** [comment]
Please can you provide the original input PNG image.

**Comment by tuif** [comment]
For sharp@0.23.4 which produces the correct result, webp version is 1.0.3
For sharp@0.26.0 which produces the wrong result, webp version is 1.1.0
The original PNG file is as the attachment

![12450](https://user-images.githubusercontent.com/10239308/91551260-75bff980-e95c-11ea-9b8c-cbf943c891bf.png)

**Comment by tuif** [comment]
Hello,
We did a little more test. Here is the code

const sharp = require('sharp');
const fs = require('fs');
(async function main() {
  console.log(sharp.versions.webp);
  let img = fs.readFileSync('/Users/tuif/Downloads/12450.png');
  let png = await sharp(img).raw().toBuffer();
  console.log('correct', png[0], png[1], png[2]);

  let webp = await sharp(img).webp({lossless: true}).raw().toBuffer();
  console.log('webp1', webp[0], webp[1], webp[2]);

  let buffer = await sharp(img).webp({lossless: true}).toBuffer();
  webp = await sharp(buffer).raw().toBuffer();
  console.log('webp2', webp[0], webp[1], webp[2]);
})();

The output with sharp 0.26.0 is:
1.1.0
correct 1 213 125
webp1 1 213 125
webp2 0 212 124

It's wrong only if we init a sharp instance of WebP-lossless from buffer. Is there any option that I miss during init?

**Comment by lovell** [comment]
Thanks for all the extra info.

The following two lines are equivalent, as the use of `raw` replaces the use of `webp`.
```diff
- let webp = await sharp(img).webp({lossless: true}).raw().toBuffer();
+ let webp = await sharp(img).raw().toBuffer();
```

The next step to is determine if the delta is introduced during encode or decode.

**Comment by lovell** [comment]
@jcupitt It looks like this is due to the way libvips accumulates pixel values when reading WebP images here:
 
https://github.com/libvips/libvips/blob/5f3482bff5f96ea37d3dae3232bba91ed4f5e960/libvips/foreign/webp2vips.c#L691-L695

The first frame of a WebP image, even if only a single frame image, will be "blended" onto a zero background, which can introduce the rounding errors being seen here.

https://github.com/lovell/sharp/files/5642277/2341.webp.txt

```sh
$ curl https://github.com/lovell/sharp/files/5642277/2341.webp.txt -o 2341.webp
$ convert 2341.webp txt:- | head -1
0,0: (  1,213,125) #01D57D
$ vips getpoint 2341.webp 0 0
0 212 124 
```

**Comment by lovell** [comment]
Fixed via https://github.com/libvips/libvips/pull/1918 - this will be included in the forthcoming libvips v8.10.3 / sharp v0.27.0.

**Comment by lovell** [comment]
sharp v0.27.0 now available with a prebuilt libvips v8.10.5 - thanks for reporting this.
